### DISCUSSION TITLE
React environment variables not working

### DISCUSSION THREAD

**Discussion by yeftanmaa** [vercel/community#3676] [Help] [answered]
### Summary

Hello everyone. Recently I have deployed my react app to Vercel and add environment variables in it. I had set in my code something like `process.env.ENV_VAR` so it can get connected with what I added in Vercel. But after pushing it to Github seems it didn't work properly (the value of `process.env.ENV_VAR` is undefined).

How to solve this?

### Example

await emailjs.send(process.env.EMAIL_SERVICES_KEY, process.env.EMAIL_TEMPLATE_NAME, templateParams, process.env.EMAIL_PUBLIC_KEY);

### Steps to Reproduce

`await emailjs.send(process.env.EMAIL_SERVICES_KEY, process.env.EMAIL_TEMPLATE_NAME, templateParams, process.env.EMAIL_PUBLIC_KEY);`

My public repo is https://github.com/yeftanmaa/react-chakra-web-portfolio

**Comment by amyegan** [comment]
Hey there @yeftanmaa. Did you add the environment variables in your project settings like this: https://vercel.com/docs/projects/environment-variables#declare-an-environment-variable

**Comment by yeftanmaa** [comment]
Hey @amyegan, yes I did exactly the same:

![image](https://github.com/vercel/community/assets/102526580/c7d4a5e2-180d-42d7-bea9-02dc40a38d12)

Link: https://vercel.com/yeftanma/react-chakra-web-portfolio/settings/environment-variables

**Comment by amyegan** [comment]
Are you using the variables in serverless functions, or are you using them in pages and components? Variables that are safe to expose to the browser generally need the `REACT_APP_` prefix. Otherwise, they are only available in server-side code.

**Reply by amyegan** [reply, marked as answer]
You can learn more about the `REACT_APP_` prefix and using custom environment variables in React apps at https://create-react-app.dev/docs/adding-custom-environment-variables/

> By default you will have `NODE_ENV` defined for you, and any other environment variables starting with `REACT_APP_`.
> 
> > WARNING: Do not store any secrets (such as private API keys) in your React app!
> > 
> > Environment variables are embedded into the build, meaning anyone can view them by inspecting your app's files.

**Comment by yeftanmaa** [comment]
I am using it in one of my page. Where do I need to add this prefix? In Vercel environment variables or in my code? Or both?
